lms/djangoapps/certificates/api.py

```diff
@@ -463,51 +463,66 @@ def get_active_web_certificate(course, is_preview_mode=None):
+
+    if org_id and mode:  # get template by org, mode, and key
+        org_mode_and_key_templates = active_templates.filter(
             organization_id=org_id,
-            course_key=course_key,
             mode=mode,
-            is_active=True
+            course_key=course_key
         )
-    # if don't template find by org and mode
-    if not template and org_id and mode:
-        template = CertificateTemplate.objects.filter(
+        template = get_templates_with_language_or_defaults(language, org_mode_and_key_templates)
+
+    # since no template matched that course_key, only consider templates with empty course_key
+    empty_course_key_templates = active_templates.filter(course_key=CourseKeyField.Empty)
+    if not template and org_id and mode:  # get template by org and mode
+        org_and_mode_templates = empty_course_key_templates.filter(
             organization_id=org_id,
-            course_key=CourseKeyField.Empty,
-            mode=mode,
-            is_active=True
+            mode=mode
         )
-    # if don't template find by only org
-    if not template and org_id:
-        template = CertificateTemplate.objects.filter(
+        template = get_templates_with_language_or_defaults(language, org_and_mode_templates)
+    if not template and org_id:  # get template by only org
+        org_templates = empty_course_key_templates.filter(
             organization_id=org_id,
-            course_key=CourseKeyField.Empty,
-            mode=None,
-            is_active=True
+            mode=None
         )
-    # if we still don't template find by only course mode
-    if not template and mode:
-        template = CertificateTemplate.objects.filter(
+        template = get_templates_with_language_or_defaults(language, org_templates)
+    if not template and mode:  # get template by only mode
+        mode_templates = empty_course_key_templates.filter(
             organization_id=None,
-            course_key=CourseKeyField.Empty,
-            mode=mode,
-            is_active=True
+            mode=mode
         )
-
+        template = get_templates_with_language_or_defaults(language, mode_templates)
     return template[0].template if template else None
 
 
+def get_templates_with_language_or_defaults(language, templates):
+    """
+    Returns templates that match passed in language.
+    Returns default templates If no language matches, or language passed is None
+    """
+    two_letter_language = _get_two_letter_language_code(language)
+    language_templates = templates.filter(language=two_letter_language)
+    return language_templates if language_templates else templates.filter(language=None)
+
+
+def _get_two_letter_language_code(language_code):
+    """
+    Shortens language to only first two characters (e.g. es-419 becomes es)
+    This is needed because Catalog returns locale language which is not always a 2 letter code.
+    """
+    return language_code[:2] if language_code else None
```

Comment:
How are we determining whether or not language-specific templates have been enabled for this course?

Reply:
IE, are we using https://github.com/edx/edx-platform/pull/15999/files#diff-c1225d0c89f33f3b74cb8ecf79a37bceR865 anywhere?

Reply:
The check for whether language-specific templates have been enabled is made prior to fetching language in the api call. It was added to certificates/view/webview.py in this PR https://github.com/edx/edx-platform/pull/16011